Sage/carbon
# feat(hidden-checkable-input): surface role prop FE-4348

### Proposed behaviour


HiddenCheckableInput: Surface `role` prop to support setting input `role`, if none set the `type` prop value is used.
Switch: Sets the input `role` to `switch`

### Current behaviour


Input has role of `checkbox` when rendered as part of `Switch` component
### Checklist




#### QA


### Additional context



### Testing instructions


Test voiceover in `switch` stories

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
